### ISSUE TITLE
[deliver] During Deliver process I get a Transporter error: Could not create the Java Virtual Machine

### ISSUE THREAD

**Issue by fishercraigj** [fastlane/fastlane#15323]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
I'm running the latest and greatest from fastlane on Xcode Server GM2.  Appears to be a transporter issue.  Anyone else seeing this?

```
WARN [2019-09-17 07:01:16.45]: [33m[iTMSTransporter] -Djava.ext.dirs=/usr/local/itms/java/lib/ext is not supported.  Use -classpath instead.
[0m
WARN [2019-09-17 07:01:16.45]: [33m[iTMSTransporter] Error: Could not create the Java Virtual Machine.
[0m
WARN [2019-09-17 07:01:16.45]: [33m[iTMSTransporter] Error: A fatal exception has occurred. Program will exit.
```

##### Command executed
deliver

##### Complete output when running fastlane, including the stack trace and command used

<details>
  <pre> INFO [2019-09-17 07:01:03.16]: [32m---------------------[0m
INFO [2019-09-17 07:01:03.16]: [32m--- Step: deliver ---[0m
INFO [2019-09-17 07:01:03.16]: [32m---------------------[0m
INFO [2019-09-17 07:01:03.20]: [32mSuccessfully loaded '/Users/optimusprime/Library/Caches/XCSBuilder/Bots/f069cf7240fd7449d69f7c92a1a251c4/Source/dod-ios/fastlane/Deliverfile' 📄[0m

+----------------------------+-----------------------------------+
|         Detected Values from './fastlane/Deliverfile'          |
+----------------------------+-----------------------------------+
| app_identifier             | com.mycompany.abc |
| username                   | myci@gmail.com        |
| team_name                  | MyCompany, Inc             |
| force                      | true                              |
| submit_for_review          | false                             |
| automatic_release          | false                             |
| run_precheck_before_submit | true                              |
| skip_metadata              | true                              |
| skip_app_version_update    | false                             |
| skip_screenshots           | true                              |
+----------------------------+-----------------------------------+

INFO [2019-09-17 07:01:03.20]: Login to App Store Connect (myci@gmail.com)
Reading keychain entry, because either user or password were empty
Loading session from '/Users/optimusprime/.fastlane/spaceship/myci@gmail.com/cookie'
Available session is not valid any more. Continuing with normal login.
Looking for App Store Connect Team with name My Company, Inc
Looking for App Store Connect Team with ID 1234567
INFO [2019-09-17 07:01:07.45]: Login successful

+--------------------------------------+--------------------------------------------------------------------------------------------------------------+
|                                                               [32mdeliver 2.131.0 Summary[0m                                                               |
+--------------------------------------+--------------------------------------------------------------------------------------------------------------+
| ipa                                  | /Users/optimusprime/Library/Caches/XCSBuilder/Bots/f069cf7240fd7449d69f7c92a1a251c4/Source/dod-ios/dodv3.ipa |
| screenshots_path                     | ./fastlane/screenshots                                                                                       |
| metadata_path                        | ./fastlane/metadata                                                                                          |
| app_version                          | 3.38                                                                                                         |
| username                             | myci@gmail.com                                                                                   |
| app_identifier                       | com.mycompany.abc                                                                           |
| platform                             | ios                                                                                                          |
| edit_live                            | false                                                                                                        |
| use_live_version                     | false                                                                                                        |
| skip_binary_upload                   | false                                                                                                        |
| skip_screenshots                     | true                                                                                                         |
| skip_metadata                        | true                                                                                                         |
| skip_app_version_update              | false                                                                                                        |
| force                                | true                                                                                                         |
| overwrite_screenshots                | false                                                                                                        |
| submit_for_review                    | false                                                                                                        |
| reject_if_possible                   | false                                                                                                        |
| automatic_release                    | false                                                                                                        |
| phased_release                       | false                                                                                                        |
| reset_ratings                        | false                                                                                                        |
| team_name                            | MyCompany, Inc                                                                                        |
| dev_portal_team_id                   | Z123456789                                                                                                   |
| run_precheck_before_submit           | true                                                                                                         |
| precheck_default_rule_level          | warn                                                                                                         |
| individual_metadata_items            | []                                                                                                           |
| ignore_language_directory_validation | false                                                                                                        |
| precheck_include_in_app_purchases    | true                                                                                                         |
+--------------------------------------+--------------------------------------------------------------------------------------------------------------+

INFO [2019-09-17 07:01:08.01]: Making sure the latest version on App Store Connect matches '3.38' from the ipa file...
INFO [2019-09-17 07:01:10.85]: [32m'3.38' is the latest version on App Store Connect[0m
INFO [2019-09-17 07:01:12.64]: Uploading app icon...
INFO [2019-09-17 07:01:14.92]: Uploading binary to App Store Connect
INFO [2019-09-17 07:01:16.35]: [32mWrote XML data to '/tmp/591981144.itmsp'[0m
INFO [2019-09-17 07:01:16.38]: Going to upload updated app to App Store Connect
INFO [2019-09-17 07:01:16.38]: [32mThis might take a few minutes. Please don't interrupt the script.[0m
DEBUG [2019-09-17 07:01:16.38]: /usr/local/itms/java/bin/java -Djava.ext.dirs=/usr/local/itms/java/lib/ext -XX:NewSize=2m -Xms32m -Xmx1024m -Xms1024m -Djava.awt.headless=true -Dsun.net.http.retryPost=false -jar /usr/local/itms/lib/itmstransporter-launcher.jar -m upload -u myci@gmail.com -p YourPassword -f /tmp/591981144.itmsp -t DAV -t Signiant -k 100000 2>&1
WARN [2019-09-17 07:01:16.45]: [33m[iTMSTransporter] -Djava.ext.dirs=/usr/local/itms/java/lib/ext is not supported.  Use -classpath instead.
[0m
WARN [2019-09-17 07:01:16.45]: [33m[iTMSTransporter] Error: Could not create the Java Virtual Machine.
[0m
WARN [2019-09-17 07:01:16.45]: [33m[iTMSTransporter] Error: A fatal exception has occurred. Program will exit.
[0m
INFO [2019-09-17 07:01:16.45]: iTunes Transporter output above ^
ERROR [2019-09-17 07:01:16.45]: [31mThe call to the iTMSTransporter completed with a non-zero exit status: 1. This indicates a failure.[0m
WARN [2019-09-17 07:01:16.45]: [33m[33mLane Context:[0m
INFO [2019-09-17 07:01:16.45]: {:DEFAULT_PLATFORM=>:ios, :PLATFORM_NAME=>:ios, :LANE_NAME=>"ios appstore_release", :BUILD_NUMBER=>"20190917065235", :SIGH_PROFILE_PATH=>"/Users/optimusprime/Library/Caches/XCSBuilder/Bots/f069cf7240fd7449d69f7c92a1a251c4/Source/dod-ios/AppStore_com.mycompany.abc.mobileprovision", :SIGH_PROFILE_PATHS=>["/Users/optimusprime/Library/Caches/XCSBuilder/Bots/f069cf7240fd7449d69f7c92a1a251c4/Source/dod-ios/AppStore_com.company.abc.mobileprovision"], :SIGH_UDID=>"dc39d489-4cc0-4cae-a25f-afece2b83f57", :SIGH_UUID=>"dc39d489-4cc0-4cae-a25f-afece2b83f57", :SIGH_NAME=>"com.company.abc AppStore", :SIGH_PROFILE_TYPE=>"app-store", :IPA_OUTPUT_PATH=>"/Users/optimusprime/Library/Caches/XCSBuilder/Bots/f069cf7240fd7449d69f7c92a1a251c4/Source/myapp/myapp.ipa", :DSYM_OUTPUT_PATH=>"/Users/optimusprime/Library/Caches/XCSBuilder/Bots/f069cf7240fd7449d69f7c92a1a251c4/Source/dod-ios/dodv3.app.dSYM.zip", :XCODEBUILD_ARCHIVE=>"/Users/optimusprime/Library/Developer/Xcode/Archives/2019-09-17/dodv3 2019-09-17 06.52.41.xcarchive"}
ERROR [2019-09-17 07:01:16.45]: [31mCould not upload binary to App Store Connect. Check out the error above[0m
INFO [2019-09-17 07:01:16.45]: [32mSuccessfully generated documentation at path '/Users/optimusprime/Library/Caches/XCSBuilder/Bots/f069cf7240fd7449d69f7c92a1a251c4/Source/dod-ios/fastlane/README.md'[0m

+------+---------------------------------------------------------------+-------------+
|                                  [32mfastlane summary[0m                                  |
+------+---------------------------------------------------------------+-------------+
| Step | Action                                                        | Time (in s) |
+------+---------------------------------------------------------------+-------------+
| 1    | Verifying fastlane version                                    | 0           |
| 2    | default_platform                                              | 0           |
| 3    | ../Scripts/reset_simulators.sh                                | 0           |
| 4    | Switch to ios set_build_to_time lane                          | 0           |
| 5    | increment_build_number                                        | 0           |
| 6    | set_info_plist_value                                          | 0           |
| 7    | sigh                                                          | 2           |
| 8    | gym                                                           | 469         |
| 9    | curl -F dsym=@"dodv3.app.dSYM.zip" -H "x-app-license-key: AA7 | 34          |
| 💥   | [31mdeliver[0m                                                       | 13          |
+------+---------------------------------------------------------------+-------------+

DEBUG [2019-09-17 07:01:16.46]: All plugins are up to date
ERROR [2019-09-17 07:01:16.46]: [31mfastlane finished with errors[0m

Looking for related GitHub issues on fastlane/fastlane...
Search query: Could not upload binary to App Store Connect. Check out the error above
 </pre>
</details>

### Environment

<details><summary><b>Loaded gems</b></summary>

| Gem                         | Version      |
| --------------------------- | ------------ |
| did_you_mean                | 1.1.2        |
| slack-notifier              | 2.3.2        |
| atomos                      | 0.1.3        |
| colored2                    | 3.1.2        |
| nanaimo                     | 0.2.6        |
| xcodeproj                   | 1.12.0       |
| rouge                       | 2.0.7        |
| xcpretty                    | 0.3.0        |
| terminal-notifier           | 2.0.0        |
| terminal-table              | 1.8.0        |
| plist                       | 3.5.0        |
| multipart-post              | 2.0.0        |
| word_wrap                   | 1.0.0        |
| public_suffix               | 2.0.5        |
| babosa                      | 1.0.2        |
| colored                     | 1.2          |
| highline                    | 1.7.10       |
| commander-fastlane          | 4.4.6        |
| faraday                     | 0.15.4       |
| http-cookie                 | 1.0.3        |
| faraday-cookie_jar          | 0.0.6        |
| gh_inspector                | 1.1.3        |
| mini_magick                 | 4.9.5        |
| multi_xml                   | 0.6.0        |
| security                    | 0.1.3        |
| xcpretty-travis-formatter   | 1.0.0        |
| bundler                     | 1.16.1       |
| faraday_middleware          | 0.13.1       |
| jwt                         | 2.1.0        |
| uber                        | 0.1.0        |
| declarative                 | 0.0.10       |
| declarative-option          | 0.1.0        |
| representable               | 3.0.4        |
| retriable                   | 3.1.2        |
| multi_json                  | 1.13.1       |
| signet                      | 0.11.0       |
| memoist                     | 0.16.0       |
| googleauth                  | 0.6.7        |
| httpclient                  | 2.8.3        |
| google-api-client           | 0.23.9       |
| digest-crc                  | 0.4.1        |
| google-cloud-storage        | 1.16.0       |
| emoji_regex                 | 1.0.1        |
| tty-cursor                  | 0.7.0        |
| tty-spinner                 | 0.9.1        |
| tty-screen                  | 0.7.0        |
| openssl                     | 2.1.0        |
| json                        | 2.2.0        |
| io-console                  | 0.4.6        |
| excon                       | 0.66.0       |
| rubyzip                     | 1.2.4        |
| CFPropertyList              | 3.0.1        |
| claide                      | 1.0.3        |
| unf                         | 0.1.4        |
| domain_name                 | 0.5.20190701 |
| fastimage                   | 2.1.7        |
| unicode-display_width       | 1.6.0        |
| addressable                 | 2.7.0        |
| os                          | 1.0.1        |
| psych                       | 3.0.2        |
| mime-types                  | 3.3          |
| mime-types-data             | 3.2019.0904  |
| xctest_list                 | 1.1.7        |
| fastlane-plugin-test_center | 3.6.3        |
</details>

**Comment by zolhaj** [comment]
Yep, running into exactly the same issue.

My understanding is that GM2's version of iTMSTransporter uses Java 11 (which removes support for the -Djava.ext.dirs argument) whereas pilot uses Java 8 (which sets the -Djava.ext.dirs argument [here](https://github.com/fastlane/fastlane/blob/049fe3dd417955c28c4365640c89c0db565ac663/fastlane_core/lib/fastlane_core/itunes_transporter.rb#L245))

We were still able to upload manually using the old iTMSTransporter

**Comment by fishercraigj** [comment]
@zolhaj - Looks like we're both exactly on the same page.  Just wish I knew the solution!

**Comment by fishercraigj** [comment]
@zolhaj - Did you app make it through the "processing" part of the manual submission?  Mine just got kicked back with a super helpful error message.  "There was an error importing this build."

**Comment by janpio** [comment]
> My understanding is that GM2's version of iTMSTransporter uses Java 11 (which removes support for the -Djava.ext.dirs argument) whereas pilot uses Java 8 (which sets the -Djava.ext.dirs argument [here](https://github.com/fastlane/fastlane/blob/049fe3dd417955c28c4365640c89c0db565ac663/fastlane_core/lib/fastlane_core/itunes_transporter.rb#L245))

Ugh, thanks @zolhaj for that information. Do you maybe also have an idea how to work around that? We will probably have to route the Xcode version information (or Java version information) in there somehow and modify the parameter/option based on that. But to what?

**Comment by zolhaj** [comment]
@fishercraigj Sorry, not sure about that one!

@janpio Our quick workaround was to use a (Java 8/9 compatible) version 1.11.0 of the Transporter from an old version of Application Loader, instead of version 2.0.0 - we copied the contents of `Application Loader.app/Contents/itms` into our itms location (which was `/usr/local/itms`)

Once our iOS 13 fires have been put out I'll try and have a look at the `iTMSTransporter.cmd` file and see how they're setting Java up. They've included Java 11 in version 2.0.0 so it should be easier to use.

**Comment by fishercraigj** [comment]
@zolhaj - Looks like the error was a false positive or something strange.  Apple rejected it for some unknown reason so I kicked another build.  Then magically, both the new build and the rejected build, entered the "processing" state and both simultaneously made it through successfully through the processing.

Moral - Apple is having some serious issues with AppStoreConnect.  I spent half the day with them on the phone about this and a few other things.  They sounded confused as all get up.

**Comment by phonggenix** [comment]
I'm encountering the same issue. On my build machine:

openjdk version "1.8.0_212"
OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_212-b03)
OpenJDK 64-Bit Server VM (AdoptOpenJDK)(build 25.212-b03, mixed mode)

**Comment by phonggenix** [comment]
I'm encountering the same issue. My remote build machine just has XCode updated to verion 11.0.
Fortunately I still have XCode 10.3 on my machine, and following @zolhaj idea, I've successfully worked around this at the moment.

**Comment by m1m6** [comment]
same issue here, and i'm new to these stuff, can someone point me exactly what should i do to overcome this issue?

**Comment by phonggenix** [comment]
Here is the link to the old ITMS (XCode 10.3) in case anyone needs:
https://drive.google.com/file/d/1AFHT-39yWkBf4f1eg87A_tyQWuCsRYsm/view?usp=sharing

Download, extract it, and override it to the /usr/local/itms folder.

Eventually I hope someone will help fix this into fastlane soon.

**Comment by m1m6** [comment]
Thanks @phonggenix you saved my day, it works!

**Comment by jason-evt** [comment]
any one still seeing the issue? I am on fastlane 2.133.0 and still seeing the error:

```
[10:27:17]: [iTMSTransporter] -Djava.ext.dirs=/usr/local/itms/java/lib/ext is not supported.  Use -classpath instead.

[10:27:17]: [iTMSTransporter] Error: Could not create the Java Virtual Machine.

[10:27:17]: [iTMSTransporter] Error: A fatal exception has occurred. Program will exit.

[10:27:17]: iTunes Transporter output above ^
[10:27:17]: The call to the iTMSTransporter completed with a non-zero exit status: 1. This indicates a failure.
```

**Comment by lilyball** [comment]
Given that Xcode 11 is out I'm rather surprised there isn't more urgency around this issue.

**Comment by sullom101** [comment]
I'm getting the same error again. I could fix it last week by downgrading xcode to 10.0

**Comment by JeonGuKang** [comment]
> 여기를



> Yep, running into exactly the same issue.
> 
> My understanding is that GM2's version of iTMSTransporter uses Java 11 (which removes support for the -Djava.ext.dirs argument) whereas pilot uses Java 8 (which sets the -Djava.ext.dirs argument [here](https://github.com/fastlane/fastlane/blob/049fe3dd417955c28c4365640c89c0db565ac663/fastlane_core/lib/fastlane_core/itunes_transporter.rb#L245))
> 
> We were still able to upload manually using the old iTMSTransporter

Wow! you are right !!! 
Thank you !!!!

**Comment by JeonGuKang** [comment]
> I'm encountering the same issue. My remote build machine just has XCode updated to verion 11.0.
> Fortunately I still have XCode 10.3 on my machine, and following @zolhaj idea, I've successfully worked around this at the moment.

Wow! you are right !!!
Thank you !!!!

**Comment by shang1219178163** [comment]
same to me, a  solution?

**Comment by daniellizik** [comment]
Where can I download Transporter 1x? I've been searching https://developer.apple.com for a while but can only see a changelog, no download link :(

**Comment by daniellizik** [comment]
@phonggenix MVP

**Comment by AnthonyAkentiev** [comment]
@phonggenix Worked like a charm. Thx

**Comment by ungripped** [comment]
For some now forgotten reason, one of our build machines had a separate version of `itms` installed in `/usr/local` which had java 11.0.2 bundled with it:

`Xcode 11.2.1`:
``` 
> /Applications/Xcode.app/Contents/SharedFrameworks/ContentDeliveryServices.framework/Versions/A/itms/java/bin/java -version 
openjdk version "1.8.0-u131-b11"
OpenJDK Runtime Environment (build 1.8.0-u131-b11-iTMSTransporter-b02)
OpenJDK 64-Bit Server VM (build 25.71-b02, mixed mode)
```

`/usr/local`:
```
> /usr/local/itms/java/bin/java -version
openjdk version "11.0.2" 2019-01-15
OpenJDK Runtime Environment 18.9 (build 11.0.2+9)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.2+9, mixed mode)
```

Removing `/usr/local/itms` caused fast lane to use the iTMSTransporter bundled with Xcode instead without problems.

**Comment by rknell** [comment]
thanks @ungripped 

I can confirm that running `sudo mv /usr/local/itms /usr/local/itms.old` fixes the problem.

When fiddling with the ITMS transporter at /usr/local/itms previously lots of things broke and I needed to revert the change. I tested this by first running fastlane, and then manually uploading using XCode and all is well.

**Comment by alex-didus** [comment]
Run on latest fastlane 2.140.0 exactly the same issue

**Comment by chiliec** [comment]
In fastlane 2.140.0 the same 😞

**Comment by dannys42** [comment]
I'm not sure if this helps, but according to this page: https://help.apple.com/itc/transporteruserguide/en.lproj/static.html#itc803b7be80

The preferred way of executing ITMS on Xcode 11.0+ is now: xcrun iTMSTransporter

**Comment by dannys42** [comment]
FYI, I looked through the code in fastlane and it looked like a bit of work to redo transporter integration to support this xcrun approach while maintaining backward compatibility.  So just to get by, I created a workaround script.  If anyone else needs, you can find it here:
https://github.com/dannys42/fastlane_workaround

**Comment by joshdholtz** [comment]
Re-opening this and then fixing it! Sorry for not seeing it sooner 😔

**Comment by joshdholtz** [comment]
This should be fixed in `2.150.0` when its released! It will be fixed in `2.150.0.rc4` when I release that tomorrow.
